llvm_jni/modules/jni.llvm/Function.bal

```diff
@@ -0,0 +1,130 @@
+import ballerina/jballerina.java;
+
+public type FunctionType readonly & record {|
+    Type returnType;
+    Type[] paramTypes;
+|};
+
+function toLLVmFunctionType(FunctionType fnType, Context context) returns handle {
+    int paramCount = fnType.paramTypes.length();
+    PointerPointer paramType = new (paramCount);
+    foreach var ty in fnType.paramTypes {
+        paramType.put(typeToLLVMType(ty));
+    }
+    return create_llvm_functionType(typeToLLVMType(fnType.returnType), paramType.jObject, paramCount, 0);
+}
+
+// LLVM C-API don't differentiate between function declarations and definitions
+public type FunctionDefn Function;
+
+public type FunctionDecl Function;
+
+public type Linkage "internal"|"external";
+
+function linkageToInt(Linkage linkage) returns int {
+    match linkage {
+        "external" => {
+            return 0;
+        }
+        "internal" => {
+            return 8;
+        }
+    }
+    panic error(string `${<string>linkage} not implemented`);
+}
+
+public type EnumAttribute "noreturn"|"cold"|"nounwind"|"readnone"|"speculatable"|"willreturn"; //FIXME: add others
+
+public distinct class Function {
+    handle LLVMFunction;
+    FunctionType fnType;
+    Context context;
+    public function init(handle llvmFunction, FunctionType fnType, Context context) {
```

Comment:
This shouldn't be public (if it takes a handle argument).

Reply:
Fixed